### ISSUE TITLE
Strange Output When Starting Server: ]1337;RemoteHost=xxx]1337;CurrentDir=/]1337;ShellIntegrationVersion=13;shell=zshv18.7.0

### ISSUE THREAD

**Issue by Songkeys** [fwouts/previewjs#1089]
### Describe the bug

In output channel:

<img width="1272" alt="image" src="https://user-images.githubusercontent.com/22665058/195181432-c18af4fb-5b7c-47d7-8c1e-edfc186acb2a.png">

```
$ node --version
]1337;RemoteHost=songkeys@songkeyss-MacBook-Pro.local]1337;CurrentDir=/]1337;ShellIntegrationVersion=13;shell=zshv18.7.0
Preview.js needs NodeJS 14.18.0+ to run.

Please upgrade then restart your IDE.
Preview.js server could not be started.
```

I tried to execute the command manually in my terminal, but the result is fine:

```
❯ /bin/zsh "-lic" "node --version"
v18.7.0
```

I also tried to use `execa` to execute, but the result is fine:

<img width="553" alt="image" src="https://user-images.githubusercontent.com/22665058/195182690-76fd1245-c308-4e35-951d-11897560069a.png">

Could it be when the env is starting, something has not fully prepared?

Possible useful info: I was using `fnm` to install node.

### Reproduction

1. Starting the extension
2. ... (I'm not sure)

### Preview.js version

v1.15.2

### Framework

\*

### System Info

- macOS m1
- vscode Version: 1.72.1 (Universal)

(`npx envinfo --system --binaries --ides --browsers` hangs quite a while on my machine.)

<img width="341" alt="image" src="https://user-images.githubusercontent.com/22665058/195183740-2cf63934-43ca-4a9e-a494-5844acec59ce.png">

### Used Package Manager

npm

**Comment by fwouts** [comment]
Hey @Songkeys, thanks for the report!

What is PS1 set to in your .zshrc? The strange characters probably come from there. Should be an easy fix once we figure it out!

**Comment by Songkeys** [comment]
Oh it may be due to the PS1 setting! I remembered I set it manually when configure my [startship](https://starship.rs/) terminal interface.

In my .zshrc:

```
PS1="%{$(iterm2_prompt_mark)%} $PS1"
```

In terminal:

```
> echo $PS1
%{%} $(/opt/homebrew/bin/starship prompt --terminal-width="$COLUMNS" --keymap="${KEYMAP:-}" --status="$STARSHIP_CMD_STATUS" --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --cmd-duration="${STARSHIP_DURATION:-}" --jobs="$STARSHIP_JOBS_COUNT")
```

After commenting my `PS1="%{$(iterm2_prompt_mark)%} $PS1"` line (for iterm2 compatibility), the issue is still the same. Guess the issue comes from the starship PS1 then?

**Comment by fwouts** [comment]
Thank you both for providing all this info!

I believe the code that needs to change is: https://github.com/fwouts/previewjs/blob/7ae8cb662562e28f286819750efa5c348ef069f0/integrations/vscode/src/start-server.ts#L159

It might even be a one-character change, simply removing the `^` so it doesn't expect the output to be on its own line... Or if I can figure out how to get it to actually split on the new line character despite broken UTF-8 before, that might be better!

I'll try and reproduce with Starship.

**Comment by fwouts** [comment]
Good news, I was able to reproduce this issue with `PS1="%{$(iterm2_prompt_mark)%} $PS1"` in `.zshrc`. Fix is on the way.

**Comment by fwouts** [comment]
The fix is now live in v1.15.3, already available for VS Code 🚀
